File: CS/B143073/GridControl.cs
using DevExpress.XtraGrid;
using DevExpress.XtraGrid.Views.Grid;
using DevExpress.XtraGrid.Registrator;
using DevExpress.XtraGrid.Views.Base;
using DevExpress.XtraGrid.FilterEditor;
using DevExpress.XtraEditors.Filtering;
using DevExpress.Utils.Menu;
using DevExpress.LookAndFeel;
using DevExpress.XtraEditors;
using System.Windows.Forms;

namespace DXSample {
    public class MyGridControl : GridControl {
        public MyGridControl() : base() { }

        protected override void RegisterAvailableViewsCore(InfoCollection collection) {
            base.RegisterAvailableViewsCore(collection);
            collection.Add(new MyGridViewInfoRegistrator());
        }
    }

    public class MyGridView : GridView {
        public MyGridView() : base() { }
        public MyGridView(GridControl grid) : base(grid) { }

        internal const string MyGridViewName = "MyGridView";
        protected override string ViewName { get { return MyGridViewName; } }

        protected override Form CreateFilterBuilderDialog(FilterColumnCollection filterColumns, 
            FilterColumn defaultFilterColumn) {
            return new MyFilterBuilder(filterColumns, GridControl.MenuManager, GridControl.LookAndFeel, this, 
                defaultFilterColumn);
        }
    }

    public class MyGridViewInfoRegistrator : GridInfoRegistrator {
        public MyGridViewInfoRegistrator() : base() { }

        public override string ViewName { get { return MyGridView.MyGridViewName; } }

        public override BaseView CreateView(GridControl grid) {
            return new MyGridView(grid);
        }
    }

    public class MyFilterBuilder : FilterBuilder {
        public MyFilterBuilder(FilterColumnCollection columns, IDXMenuManager manager, UserLookAndFeel lookAndFeel, 
            ColumnView view, FilterColumn fColumn) 
            : base(columns, manager, lookAndFeel, view, fColumn) {
            sbOK.Enabled = sbApply.Enabled = false;
            ((FilterControl)fcMain).FilterChanged += 
                new FilterChangedEventHandler(OnFilterControlFilterChanged);
        }

        private void OnFilterControlFilterChanged(object sender, FilterChangedEventArgs e) {
            sbOK.Enabled = sbApply.Enabled = true;
            ((FilterControl)fcMain).FilterChanged -= new FilterChangedEventHandler(OnFilterControlFilterChanged);
        }
    }
}

File: VB/B143073/GridControl.vb
Imports Microsoft.VisualBasic
Imports DevExpress.XtraGrid
Imports DevExpress.XtraGrid.Views.Grid
Imports DevExpress.XtraGrid.Registrator
Imports DevExpress.XtraGrid.Views.Base
Imports DevExpress.XtraGrid.FilterEditor
Imports DevExpress.XtraEditors.Filtering
Imports DevExpress.Utils.Menu
Imports DevExpress.LookAndFeel
Imports DevExpress.XtraEditors
Imports System.Windows.Forms

Namespace DXSample
	Public Class MyGridControl
		Inherits GridControl
		Public Sub New()
			MyBase.New()
		End Sub

		Protected Overrides Sub RegisterAvailableViewsCore(ByVal collection As InfoCollection)
			MyBase.RegisterAvailableViewsCore(collection)
			collection.Add(New MyGridViewInfoRegistrator())
		End Sub
	End Class

	Public Class MyGridView
		Inherits GridView
		Public Sub New()
			MyBase.New()
		End Sub
		Public Sub New(ByVal grid As GridControl)
			MyBase.New(grid)
		End Sub

		Friend Const MyGridViewName As String = "MyGridView"
		Protected Overrides ReadOnly Property ViewName() As String
			Get
				Return MyGridViewName
			End Get
		End Property

		Protected Overrides Function CreateFilterBuilderDialog(ByVal filterColumns As FilterColumnCollection, ByVal defaultFilterColumn As FilterColumn) As Form
			Return New MyFilterBuilder(filterColumns, GridControl.MenuManager, GridControl.LookAndFeel, Me, defaultFilterColumn)
		End Function
	End Class

	Public Class MyGridViewInfoRegistrator
		Inherits GridInfoRegistrator
		Public Sub New()
			MyBase.New()
		End Sub

		Public Overrides ReadOnly Property ViewName() As String
			Get
				Return MyGridView.MyGridViewName
			End Get
		End Property

		Public Overrides Function CreateView(ByVal grid As GridControl) As BaseView
			Return New MyGridView(grid)
		End Function
	End Class

	Public Class MyFilterBuilder
		Inherits FilterBuilder
		Public Sub New(ByVal columns As FilterColumnCollection, ByVal manager As IDXMenuManager, ByVal lookAndFeel As UserLookAndFeel, ByVal view As ColumnView, ByVal fColumn As FilterColumn)
			MyBase.New(columns, manager, lookAndFeel, view, fColumn)
			sbApply.Enabled = False
			sbOK.Enabled = sbApply.Enabled
			AddHandler (CType(fcMain, FilterControl)).FilterChanged, AddressOf OnFilterControlFilterChanged
		End Sub

		Private Sub OnFilterControlFilterChanged(ByVal sender As Object, ByVal e As FilterChangedEventArgs)
			sbApply.Enabled = True
			sbOK.Enabled = sbApply.Enabled
			RemoveHandler (CType(fcMain, FilterControl)).FilterChanged, AddressOf OnFilterControlFilterChanged
		End Sub
	End Class
End Namespace